db/db.go

```diff
@@ -0,0 +1,24 @@
+package db
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/mount-joy/thelist-lambda/data"
+)
+
+// DB - interface for talking to the database
+type DB interface {
+	GetItemsOnList(listID *string) (*[]data.Item, error)
```

Comment:
I think this is just a taste thing, but I prefer it when the interface doesn't state the name of the variable(s)

Reply:
No clear why this would take a pointer to string, and return a pointer to item

Reply:
> No clear why this would take a pointer to string, and return a pointer to item

As in you you're asking why it isn't:

```suggestion
	GetItemsOnList(listID string) ([]data.Item, error)
```